pkg/util/traceid.go

```diff
@@ -51,6 +51,14 @@ func TraceIDToHexString(byteID []byte) string {
 	return id
 }
 
+// SpanIDToHexString converts a span ID to its string representation and WITHOUT removing any leading zeros.
+// Assuming everything passed in here will be 64 bit or less
+func SpanIDToHexString(byteID []byte) string {
+	id := hex.EncodeToString(byteID)
+	return id[len(id)-16:]
```

Comment:
so even though a span should never be less than 16 characters this function should not assume that. it should protect itself from invalid input. 

as written with a little bit of bad luck or a bad assumption somewhere else this would panic tempo components.